torch/_higher_order_ops/cast_symbool_to_symint.py

```diff
@@ -0,0 +1,69 @@
+import sympy
+
+import torch
```

Comment:
Can you briefly explain why this needs to be a higher order op.

Reply:
The main reason is we insert a casting op from symbool to symint before calling into the extracted graph, we want to be able to trace the casting into the graph. Otherwise, the create proxy call in a downstream higher order op e.g. cond, cannot find the associated proxy for the SymInt and errors [here](https://github.com/pytorch/pytorch/blob/main/torch/fx/experimental/proxy_tensor.py#L459)